# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/qemu/hotfix-uio.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- qemu-6.2.0-rc0/include/qemu/osdep.h.vanilla	2021-11-16 18:05:10.000000000 +0100
+++ qemu-6.2.0-rc0/include/qemu/osdep.h	2021-11-16 18:14:55.000000000 +0100
@@ -58,6 +58,7 @@
 #include <stdlib.h>
 #undef daemon
 QEMU_EXTERN_C int daemon(int, int);
+#include <unistd.h>
 #endif
 
 #ifdef _WIN32
@@ -104,7 +105,7 @@
 #include <setjmp.h>
 #include <signal.h>
 
-#ifdef CONFIG_IOVEC
+#if defined(CONFIG_IOVEC) || defined(__APPLE__)
 #include <sys/uio.h>
 #endif
 
@@ -606,7 +606,7 @@
 
 int qemu_get_thread_id(void);
 
-#ifndef CONFIG_IOVEC
+#if !defined(CONFIG_IOVEC) && !defined(__APPLE__)
 struct iovec {
     void *iov_base;
     size_t iov_len;
